### DISCUSSION TITLE
Renaming bookmarks

### DISCUSSION THREAD

**Discussion by engeir** [ahrm/sioyek#801] [Q&A] [answered]
I am currently trying to write a python script using the `sioyek` package to rename bookmarks, also discussed in [this issue](https://github.com/ahrm/sioyek/issues/372). It feels like it should not be too hard, but I struggle to figure out how to find the closest bookmark.

That is, I assume the `delete_bookmark` command will always delete whichever bookmark is closer, so the idea would be to first find the closest bookmark, copy the description, delete it and insert a new bookmark where the description is copied into the insert field.

One caveat to this is that `add_bookmark` in sioyek opens an input field, while either of the following three lines will just create new bookmarks without the option to specify the bookmark name/description.

```python
Sioyek().add_bookmark("this is a bookmark made from sioyek python iterface")
subprocess.run([SIOYEK_PATH, "--execute-command", "add_bookmark"])
subprocess.run([SIOYEK_PATH, "--execute-command", "add_bookmark", "--execute-command-data",  "this is a bookmark made from command line"])
```

Any tips on how to move forward with this?

**Comment by ahrm** [comment, marked as answer]
Unfortunately, in the current version of sioyek there is no way to fill in a text input from python without executing the command. It will be available in the next version though.

**Reply by engeir** [reply]
I noticed that calling the `add_bookmark` command after having copied text inside sioyek (i.e. with `<C-c>`) this text was inserted in the text field and can be edited.

So, as a workaround for now, would finding the closest bookmark, copy the text and then deleting it be possible? Because in this case I suppose it would be possible to right after call `add_command` manually in sioyek and then have the deleted bookmark description available.

EDIT:
Ah, it was just because I did not remove the highlight of the text I just copied before calling `add_bookmark`. So the copying had nothing to do with it after all.
